### PR TITLE
Conf state trx + Change mongodb get result for key not found  + Fix byte[] in cosmosdb

### PR DESCRIPTION
**Pull request by artursouza**
# Description

Validation in this run:
https://github.com/dapr/components-contrib/runs/1868788838?check_suite_focus=true

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #407 
Will also close: #684 
Will also close: #685

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [X] Code compiles correctly
* [X] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: state/mongodb/mongodb.go [hunk 1]
```diff
@@ -167,6 +167,11 @@ func (m *MongoDB) Get(req *state.GetRequest) (*state.GetResponse, error) {
 	filter := bson.M{id: req.Key}
 	err := m.collection.FindOne(ctx, filter).Decode(&result)
 	if err != nil {
+		if err.Error() == "mongo: no documents in result" {
```
Comment (mukundansundar) [inline state/mongodb/mongodb.go:170]: @yaron2 @artursouza Isn't this a breaking change? 
My question is do we want to change it this way now? 
Should the code be changed specifically for conf-tests? Or should it be handled as a special case in conf-tests?

Comment (yaron2) [reply state/mongodb/mongodb.go:170]: If this is a breaking change for the component, this is all the more reason to do this now.

Comment (artursouza) [reply state/mongodb/mongodb.go:170]: We better break it now then later.

File: state/redis/redis.go [hunk 2]
```diff
@@ -374,3 +372,17 @@ func (r *StateStore) parseETag(req *state.SetRequest) (int, error) {
...
+	if value == nil {
+		return nil
+	}
+
+	s, ok := value.(string)
+	if ok {
+		return []byte(s)
+	}
+
+	bt, _ := utils.Marshal(value, r.json.Marshal)
```
Comment (mukundansundar) [inline state/redis/redis.go:386]: propagate err from here ...

**Review by yaron2 — APPROVED** [review]

**Pull request merged**